### PR TITLE
 Initialize repo with Prometheus middleware

### PR DESCRIPTION
**Pull request by ckarenz**

### CODE REVIEW

File: prometheus.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	requestsOpts := prometheus.CounterOpts{
+		Name:        requestsMetricName,
+		Help:        "Count of HTTP requests, partitioned by status code, method and HTTP path.",
+		ConstLabels: labels,
+	}
+	requests := prometheus.NewCounterVec(requestsOpts, partitions)
+	registry.MustRegister(requests)
+
+	latenciesOpts := prometheus.HistogramOpts{
+		Name:        "latencies_seconds",
```
Comment (ckarenz) [inline prometheus.go]: This is a change in behavior. I can multiply everything by 1000 and change to milliseconds if preferred.

Comment (viveklak) [reply prometheus.go]: I think we should go with milliseconds.

File: prometheus_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-chi/chi"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
```
Comment (viveklak) [inline prometheus_test.go]: Curious how you chose `dto` as the alias here :)?

Comment (ckarenz) [reply prometheus_test.go]: It's the alias Prometheus uses internally. I'm not sure why.  I'll change it, though.

File: prometheus.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+			next.ServeHTTP(ww, r)
+
+			// Use the route pattern instead of URL to avoid explosion in metric
+			// dimensionality. The route pattern should be determined after the
+			// router chain is complete, meaning this should run after 'next'.
+			code := strconv.Itoa(ww.Status())
+			path := chi.RouteContext(r.Context()).RoutePattern()
+			requests.WithLabelValues(code, r.Method, path).Inc()
+			latencies.WithLabelValues(code, r.Method, path).Observe(time.Since(start).Seconds() * 1000)
```
Comment (viveklak) [inline prometheus.go]: Missed this somehow. I think we are loosing precision here. We want the math in nanoseconds then divide by 1000000 instead

Comment (ckarenz) [reply prometheus.go]: `Seconds()` returns full precision.  I'm assuming you're not concerned about FP rounding error, but loss of sub-second precision.

```golang
func (d Duration) Seconds() float64 {
	sec := d / Second
	nsec := d % Second
	return float64(sec) + float64(nsec)/1e9
}
```

**Pull request merged**